lib/network_utils.ml

```diff
@@ -806,16 +807,15 @@ module Linux_bonding = struct
             Sysfs.read_one_line (Sysfs.getpath master ("bonding/" ^ prop))
           in
           if prop = "mode" then
-            Some
-              ( prop
-              , List.hd (Astring.String.cuts ~empty:false ~sep:" " bond_prop) )
+            Astring.String.cut ~sep:" " (String.trim bond_prop)
+            |> Option.map (fun (mode, _) -> (prop, mode))
```

Comment:
Now the debug line won't print if the List.hd fails, I don't think it was meant to be printed.

Reply:
I think we only need the `try` around `Sysfs.read_one_line` now

Reply:
The docs for `cut` say that it returns "None if sep can't be matched in s". So if `bond_prop` does not contain a space, then the result will now be `None`.

For `cuts`, the docs say: "If there is no separator match in s, the list [s] is returned.". So the result would be `Some (prop, bond_prop)`

Reply:
The original took the first "word" of the line, ignoring leading spaces.

I think this can be done without using Optional:
```ocaml
let get_mode line =
  let a_space char = char = ' ' in
  Astring.String.(
    line |> drop ~sat:a_space |> take ~sat:(Fun.negate a_space))
in
```